src/api/userAPI.ts

```diff
@@ -1,29 +1,38 @@
+import { BASE_URL } from '@env';
 import jwt_decode from 'jwt-decode';
 
 import { UserType } from '../types/types';
 
 import { host } from './networkClient';
 
 interface ILogin {
-   token: string
+   token: string,
 }
 
-export const registration = async (email, password): Promise<UserType | void> => {
+export const registration = async (email, password): Promise<UserType | number | void> => {
   try {
     const { data } = await host.post<ILogin>('api/user/registration', { email, password });
 
     return jwt_decode(data.token);
   } catch (e) {
-    console.log(e);
+    if ((e as Error).message === 'Network Error') {
+      return 500;
+    }
+
+    return 400;
```

Comment:
this is wrong, the correct error code must be returned from BE and you have to get it from here like return e.statusCode (or something like that)